apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/AbstractCleanupJob.java

```diff
@@ -133,24 +127,21 @@ private int executeDeleteTransaction(List<Query> queries, TransactionManager tm)
    * @param tm the TransactionManager to use to create queries.
    * @return the list of queries.
    */
-  private List<Query> buildDeleteQueries(Instant cutoffDate, TransactionManager tm) {
-    List<Query> queries = new ArrayList<>();
-    String parentTableName = getTableNames().getLast();
+  private Query buildDeleteQueries(TransactionManager tm) {
+    AtomicReference<Query> atomicQuery = new AtomicReference<>();
```

Comment:
Do we need the `AtomicReference` here? Is there some race condition we're trying to prevent?

Reply:
Since the String is referenced inside of a lambda, it must be final. Using an AtomicReference is a way to do that. Another way would be putting into an array with a single element.

Reply:
Huh, interesting. Good to know.

Reply:
Actually, why do we need to call `executeProcedure` in the first place? It doesn't look like we're running any queries inside the callback here. Can't we just do `tm.getEntityManager().createNativeQuery(queryStr)`?

Reply:
Let me try it